third_party/amd/lib/TritonAMDGPUToLLVM/OptimizeLDSUsage.cpp

```diff
@@ -97,6 +97,8 @@ class OptimizeAMDLDSUsage
 
     auto ctx = srcEnc.getContext();
     auto rank = srcType.getShape().size();
+    if (rank < 2)
```

Comment:
I suggest to add a comment, something like "1d tensors are small and there are not enough parameters in layout that could be varied to achieve sufficient LDS reduction"